test/test_pipeline.py

```diff
@@ -182,203 +204,320 @@ def test_to_code():
-    )
+
+def test_get_config_creates_two_different_dependent_components_of_same_type():
+    child_a = ChildComponent(some_key="A")
+    child_b = ChildComponent(some_key="B")
+    parent = ParentComponent(dependent=child_a)
+    parent2 = ParentComponent(dependent=child_b)
     p_ensemble = Pipeline()
-    p_ensemble.add_node(component=es_retriever, name="EsRetriever", inputs=["Query"])
-    p_ensemble.add_node(component=emb_retriever, name="EmbeddingRetriever", inputs=["Query"])
-    p_ensemble.add_node(
-        component=JoinDocuments(join_mode="merge"), name="JoinResults", inputs=["EsRetriever", "EmbeddingRetriever"]
-    )
+    p_ensemble.add_node(component=parent, name="ParentA", inputs=["Query"])
+    p_ensemble.add_node(component=parent2, name="ParentB", inputs=["Query"])
+    p_ensemble.add_node(component=JoinDocuments(join_mode="merge"), name="JoinResults", inputs=["ParentA", "ParentB"])
+
+    expected_components = [
+        {"name": "ParentA", "type": "ParentComponent", "params": {"dependent": "ChildComponent"}},
+        {"name": "ChildComponent", "type": "ChildComponent", "params": {"some_key": "A"}},
+        {"name": "ParentB", "type": "ParentComponent", "params": {"dependent": "ChildComponent_2"}},
+        {"name": "ChildComponent_2", "type": "ChildComponent", "params": {"some_key": "B"}},
+        {"name": "JoinResults", "type": "JoinDocuments", "params": {"join_mode": "merge"}},
+    ]
 
-    code = _PipelineCodeGen.generate_code(pipeline=p_ensemble, pipeline_variable_name="p", generate_imports=False)
+    expected_pipelines = [
+        {
+            "name": "query",
+            "type": "Pipeline",
+            "nodes": [
+                {"name": "ParentA", "inputs": ["Query"]},
+                {"name": "ParentB", "inputs": ["Query"]},
+                {"name": "JoinResults", "inputs": ["ParentA", "ParentB"]},
+            ],
+        }
+    ]
+
+    config = p_ensemble.get_config()
+    for expected_pipeline in expected_pipelines:
+        assert expected_pipeline in config["pipelines"]
+    for expected_component in expected_components:
+        assert expected_component in config["components"]
+
+
+def test_PipelineCodeGen_simple_pipeline():
+    config = {
+        "components": [
+            {
+                "name": "retri",
+                "type": "ElasticsearchRetriever",
+                "params": {"document_store": "ElasticsearchDocumentStore", "top_k": 20},
+            },
+            {
+                "name": "ElasticsearchDocumentStore",
+                "type": "ElasticsearchDocumentStore",
+                "params": {"index": "my-index"},
+            },
+        ],
+        "pipelines": [{"name": "query", "type": "Pipeline", "nodes": [{"name": "retri", "inputs": ["Query"]}]}],
+    }
+
+    code = _PipelineCodeGen.generate_code(pipeline_config=config, pipeline_variable_name="p", generate_imports=False)
     assert code == (
```

Comment:
I know it's debatable, but I would not test for string equality here. Adding or removing whitespace from the implementation can break this test, for example.

There are a few alternatives:
- Check for all the lines to be in the output and in the correct order
- Execute the code and validate that what you expect appeared in the local scope, like you did with the integration test. Components are mocked anyway, this will stay small and quick.

Same for the other code generation tests where you use string equality on the output code.

Reply:
I think it's not worth the effort. If the order wasn't important ok, but here it is. So let's better be a little too specific here in favor of making the assertion more complex and maybe even less understandable if the test fails.